### PR TITLE
ATD Knack Services to Prefect 2

### PR DESCRIPTION
**Pull request by Charlie-Henry**
Moves these Prefect 1 Knack Services flows to Prefect 2:
- atd_knack_artbox_signals
- atd_knack_markings_contractor_work_orders

New Knack Services based on scripts in atd-data-deploy:
- atd_knack_street_segment_updater
- atd_knack_data_tracker_sr_asset_assign
- atd_knack_services_data_tracker_location_updater

New knack service deployment:
- atd_knack_tcp_submissions_socrata

## Secrets:

The primary storage for all knack services scripts is in the `atd-knack-services` JSON block in Prefect. This block's JSON schema is based on what knack app you are using. A special key is also set aside for storing when each flow was last executed successfully, called `PREV-EXECS`. We can also use this for defining secrets for knack service flows that pass data between knack apps.

The benefit is that we don't need one secret block per knack service to hopefully reduce the crufty blocks like: `atd_knack_tcp_submissions_socrata_secret_block`
```
{
	"data-tracker":
	{
		"secret_1":"abc",
		"secret_2":"xyz"
	},
	"signs-markings":
	{
		"secret_3":"def",
		"secret_4":"uvw"
	},
	"PREV-EXECS":
	{
		"Knack Services 1": "2023-03-23",
    	        "Knack Services 2": "2023-03-28"
	}
}
```

Issues
- https://github.com/cityofaustin/atd-data-tech/issues/12070
- https://github.com/cityofaustin/atd-data-tech/issues/11769
- https://github.com/cityofaustin/atd-data-tech/issues/10901
- https://github.com/cityofaustin/atd-data-tech/issues/11831
- https://github.com/cityofaustin/atd-data-tech/issues/12081
- https://github.com/cityofaustin/atd-data-tech/issues/11637

### CODE REVIEW

**Review by frankhereford — APPROVED** [review]
Charlie, this is a chunk of work! Thank you! I have reviewed the code, and it looks super clean and top shelf. I have not attempted to verify the flows by running them, but please let me know if you'd like that, and I'll be happy to come back and do it.

Thanks again!

File: flows/atd-knack-services/atd_knack_tcp_submissions_socrata.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+@task(
+    name="update_exec_date",
+    retries=10,
+    retry_delay_seconds=timedelta(seconds=15).seconds,
+)
+def update_exec_date(json_block):
+    # Update our JSON block with the updated date of last flow execution
+    block = JSON.load(json_block)
+    block.value["PREV_EXECS"][FLOW_NAME] = datetime.today().strftime("%Y-%m-%d")
+    block.save(name=json_block, overwrite=True)
```
[the remark this answers is not in the document]
Comment (chiaberry) [reply flows/atd-knack-services/atd_knack_tcp_submissions_socrata.py:117]: The way we do it in airflow now is accessing the prev_execution_date_success that I believe comes out of the box of at least airflow 1

example: https://github.com/cityofaustin/atd-airflow/blob/master/dags/atd_knack_dms.py#L52

File: flows/atd-knack-services/__pycache__/atd_knack_data_tracker_sr_asset_assign.cpython-39.pyc
Comment (chiaberry) [inline flows/atd-knack-services/__pycache__/atd_knack_data_tracker_sr_asset_assign.cpython-39.pyc:1]: does this need to be included in the PR?

File: flows/atd-knack-services/atd_knack_tcp_submissions_socrata.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnclary) [reply flows/atd-knack-services/atd_knack_tcp_submissions_socrata.py:117]: yeah! airflow calls them [template variables](https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html).

[the remark this answers is not in the document]
Comment (frankhereford) [reply flows/atd-knack-services/atd_knack_tcp_submissions_socrata.py:117]: Ah, how perfect! TIL - thanks y'all!

**Pull request merged**